internal/testpeertls/revocations.go

```diff
@@ -8,50 +8,90 @@ import (
 	"crypto/x509"
 	"crypto/x509/pkix"
 
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/identity"
 	"storj.io/storj/pkg/peertls"
 	"storj.io/storj/pkg/peertls/extensions"
-	"storj.io/storj/pkg/pkcrypto"
 )
 
 // RevokeLeaf revokes the leaf certificate in the passed chain and replaces it
 // with a "revoking" certificate, which contains a revocation extension recording
 // this action.
-func RevokeLeaf(keys []crypto.PrivateKey, chain []*x509.Certificate) ([]*x509.Certificate, pkix.Extension, error) {
-	var revocation pkix.Extension
-	revokingKey, err := pkcrypto.GeneratePrivateKey()
-	if err != nil {
-		return nil, revocation, err
+func RevokeLeaf(caKey crypto.PrivateKey, chain []*x509.Certificate) ([]*x509.Certificate, pkix.Extension, error) {
+	if len(chain) < 2 {
+		return nil, pkix.Extension{}, errs.New("revoking leaf implies a CA exists; chain too short")
```

Comment:
nit - could maybe help to make a zeebo errs "revocations" `Error` class, and then make this `Error.New`? not critical though

Reply:
Good idea. I was being lazy because it was a test helper :p